libs/langchain/langchain/chains/query_constructor/ir.py

```diff
@@ -102,6 +102,7 @@ class Comparison(FilterDirective):
     comparator: Comparator
     attribute: str
     value: Any
+    value_type: str
```

Comment:
probably should add a default value here for backwards compat

Reply:
What if you keep this schema exactly as it is right now, but allow `Value` to be of a dict type, the dict type can have { 'type': ..., 'value": } inside it? This means that can still use python primitive types easily without needing to carry an additional value_type, and can describe more complex objects with a dict? Does bad happen if we use that representation?

Reply:
we need to update all the vector store translators